### PR TITLE
vsphere_guest: Added support for resizing disks when state is 'reconfigured'

### PR DESCRIPTION
**Pull request by sysreq0**
This pull request adds support for increasing the size of one or multiple disks (vSphere's API doesn't seem to allow decreasing the size of a disk) whenever the 'reconfigured' state is used.

### CODE REVIEW

**Comment by robynbergeron** [comment]
Thanks @sysreq0. @chrrrles @rhoop please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -858,6 +858,50 @@ def reconfigure_vm(vsphere_client, vm, module, esxi, resource_pool, cluster_name
...
+            try:
+                disksize = int(vm_disk[disk]['size_gb'])
+                # Convert the disk size to kilobytes
+                disksize = disksize * 1024 * 1024
+            except (KeyError, ValueError):
+                vsphere_client.disconnect()
+                module.fail_json(msg="Error in '%s' definition. Size needs to be specified as an integer." % disk)
+
+            # Get a list of the hard drives 
+            dev_list = [d for d in vm.properties.config.hardware.device if d._type=='VirtualDisk']
```
Comment (chrrrles) [inline cloud/vmware/vsphere_guest.py]: It may be more efficient to create `dev_list` prior to the `for disk in...` loop.

**Comment by sysreq0** [comment]
Thanks for the feedback, @chrrrles. I have made the appropriate changes to the pull request.

**Comment by juliedavila** [comment]
shipit.

**Comment by mtward** [comment]
Could this be extended to add disks during reconfigured state?  That is how we would most often use it.

**Comment by juliedavila** [comment]
@mtward probably, but would need to be a new pull request

**Comment by anolan622** [comment]
has anyone gotten this feature set in?  I am trying to do a reconfigure with disksize to either grow disk and/or add a disk.

any luck?

**Pull request merged**